wandb/sdk/launch/runner/kubernetes_monitor.py

```diff
@@ -116,200 +93,256 @@ def _is_container_creating(status: "V1PodStatus") -> bool:
-        self,
-        job_field_selector: str,
-        pod_label_selector: str,
-        namespace: str,
-        batch_api: "BatchV1Api",
-        core_api: "CoreV1Api",
-        custom_api: "CustomObjectsApi" = None,
-        group: Optional[str] = None,
-        version: Optional[str] = None,
-        plural: Optional[str] = None,
-    ) -> None:
-        """Initialize KubernetesRunMonitor.
-
-        If a custom api is provided, the group, version, and plural arguments must also
-        be provided. These are used to query the custom api for a launched custom
-        object (CRD). Group, version, and plural in this context refer to the
-        Kubernetes API group, version, and plural for the CRD. For more information
-        see: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/
-
-        The run monitor starts two threads to watch for pods and jobs/crds matching the
-        provided selectors. The status is set to "starting" when the run monitor is
-        initialized. The status is set to "running" when a pod matching the pod selector
-        is found with a status of "Running" or has a container with a status of
-        "ContainerCreating". The status is set to "finished" when a job matching the job
-        selector is found with a status of "Succeeded". The status is set to "failed"
-        when a job matching the job selector is found with a status of "Failed" or a pod
-        matching the pod selector is found with a status of "Failed". The status is set
-        to "preempted" when a pod matching the pod selector is found with a condition
-        type of "DisruptionTarget" and a reason of "EvictionByEvictionAPI",
-        "PreemptionByScheduler", or "TerminationByKubelet".
-
-        The logic for the CRD is similar to the logic for the job, but we inspect
-        both the phase of the CRD and the conditions since some CRDs do not have a
-        phase field.
-
-        Arguments:
-            job_field_selector: The field selector for the job or crd.
-            pod_label_selector: The label selector for the pods.
-            namespace: The namespace to monitor.
-            batch_api: The batch api client.
-            core_api: The core api client.
-            custom_api: The custom api client.
-            group: The group of the CRD.
-            version: The version of the CRD.
-            plural: The plural of the CRD.
-
-        Returns:
-            None.
-        """
-        self.pod_label_selector = pod_label_selector
-        self.job_field_selector = job_field_selector
-        self.namespace = namespace
-        self.batch_api = batch_api
-        self.core_api = core_api
-        self.custom_api = custom_api
-        self.group = group
-        self.version = version
-        self.plural = plural
+def _state_from_replicated_status(status_dict: Dict[str, int]) -> Optional[State]:
+    """Get the status from the replicated jobs status.
```

Comment:
Maybe more clear to say:
Determines the state based on the readiness and activity of the pod